stdlib/2and3/fractions.pyi

```diff
@@ -0,0 +1,89 @@
+# Stubs for fractions
+# See https://docs.python.org/3/library/fractions.html
+
+from typing import Optional, TypeVar, Union, overload
+from numbers import Complex, Rational
+from decimal import Decimal
+import sys
+
+_TComplex = TypeVar('_TComplex', bound=Complex)
+_TOtherNum = TypeVar('_TOtherNum', int, float, Decimal)
+_TNum = TypeVar('_TNum', int, float, Decimal, Complex)
+
+@overload
+def gcd(a: _TComplex, b: _TComplex) -> _TComplex: ...
```

Comment:
Hmm, so it looks like the return type of gcd here ultimately depends on what the type of `a % b` is, which makes the gcd function more complicated then I was originally expecting.

Some things I observed:

- `numbers.Complex` does not have any kind of dunder methods related to mods, so shouldn't be a legal input type in either Python 2 or Python 3.
- However, the builtins `complex` type _does_ allow you to take the mod in Python 2, but NOT in Python 3.
- If one argument is a complex number, and the other is any other type, the gcd function will either throw a TypeError or will enter an infinite loop (!!).

I also found the full matrix of allowable parameters to return types is this (not taking into account whatever any custom number subclasses might do):

```
            | int          float        Fraction     Decimal      complex
------------+ -----------  -----------  -----------  -----------  -----------
int         | int          float        Fraction     Decimal      Inf. Loop
float       | float        float        float        TypeError    Inf. Loop
Fraction    | Fraction     float        Fraction     TypeError    TypeError
Decimal     | Decimal      TypeError    TypeError    Decimal      TypeError
complex     | Inf. Loop    Inf. Loop    TypeError    TypeError    complex
```

Thankfully, the diagonal (where both parameters are the same type) appear to be relatively sane.

So, given all of the above, and given that `fractions.gcd` is a deprecated function anyways as of Python 3 (people are supposed to use math.gcd instead, which accepts only ints), would you rather I:

- Attempt to come up with a system of overloads to handle the full matrix.
- Attempt to handle the case where both parameters are the same type, and just provide separate annotations for Python 2 and 3 to handle complex numbers.
- Just disallow allowing complex numbers as parameters completely on the grounds that nobody is actually going to do that/they're going to be using cmath instead.
- Don't bother providing a type signature, and just make the input and output types Any.
- Something else?